### PR TITLE
Fix 11 unittests by autofix

### PR DESCRIPTION
**Pull request by RedContritio**
### PR Docs

更新添加修复了部分 api 的单测，目前不完善的单测余量 123 个，具体情况如下：

表格含义：
1. 五列分别是 api、全部位置参数、全部关键字参数、乱序关键字参数、有默认值的全部默认值。
2. ❓表示存在部分尚未支持的参数/单测，🔁 表示存在重载形式，🟢 表示存在 cornercase （如 `torch.Tensor.type` 需要处理其他 api 的 fall through）
3. ✅ 表示有对应的单测，❌ 表示无对应单测，⚠️ 表示该类型单测无法检查。

| api | all args | all kwargs | kwargs out of order | all default |
| --- | --- | --- | --- | --- |
| torch.Tensor.contiguous | ✅ | ❌ | ✅ | ✅ |
| torch.Tensor.detach_ | ✅ | ✅ | ✅ | ⚠️ |
| torch.Tensor.is_contiguous | ✅ | ❌ | ✅ | ✅ |
| 🔁 torch.Tensor.max | ❌ | ❌ | ❌ | ✅ |
| ❓ torch.Tensor.median | ⚠️ | ❌ | ✅ | ✅ |
| 🔁 torch.Tensor.min | ❌ | ❌ | ❌ | ✅ |
| ❓ torch.Tensor.nanmedian | ⚠️ | ❌ | ✅ | ✅ |
| torch.Tensor.new_empty | ❌ | ✅ | ✅ | ✅ |
| torch.Tensor.new_tensor | ✅ | ❌ | ✅ | ✅ |
| torch.Tensor.new_zeros | ❌ | ✅ | ✅ | ✅ |
| ❓ torch.Tensor.norm | ❌ | ❌ | ❌ | ❌ |
| 🟢 torch.Tensor.remainder | ✅ | ❌ | ✅ | ✅ |
| 🔁 torch.Tensor.scatter | ❌ | ❌ | ❌ | ✅ |
| 🔁 ❓ torch.Tensor.scatter_ | ✅ | ❌ | ❌ | ✅ |
| ❓ torch.Tensor.softmax | ✅ | ✅ | ✅ | ❌ |
| 🔁 torch.Tensor.sort | ✅ | ❌ | ❌ | ✅ |
| ❓ torch.Tensor.split | ✅ | ✅ | ❌ | ✅ |
| 🔁 torch.Tensor.std | ✅ | ❌ | ❌ | ❌ |
| 🔁 torch.Tensor.to | ❌ | ❌ | ❌ | ✅ |
| 🟢 torch.Tensor.type | ❌ | ❌ | ✅ | ✅ |
| 🟢 torch.Tensor.uniform_ | ✅ | ❌ | ❌ | ✅ |
| ❓ torch.Tensor.unique | ⚠️ | ❌ | ❌ | ✅ |
| 🔁 torch.Tensor.var | ✅ | ❌ | ❌ | ⚠️ |
| 🔁 torch.Tensor.view | ✅ | ❌ | ❌ | ✅ |
| ❓ torch.autograd.grad | ⚠️ | ✅ | ✅ | ✅ |
| ❓ torch.bernoulli | ✅ | ❌ | ✅ | ⚠️ |
| ❓ torch.broadcast_shapes | ✅ | ✅ | ✅ | ❌ |
| 🟢 ❓ torch.chain_matmul | ✅ | ✅ | ✅ | ❌ |
| 🟢 torch.clamp | ✅ | ✅ | ✅ | ❌ |
| 🟢 torch.clip | ✅ | ✅ | ✅ | ❌ |
| torch.cuda.amp.autocast | ❌ | ❌ | ❌ | ✅ |
| 🔁 torch.cumulative_trapezoid | ✅ | ❌ | ❌ | ✅ |
| torch.diff | ✅ | ✅ | ✅ | ⚠️ |
| torch.distributions.AffineTransform | ❌ | ❌ | ❌ | ✅ |
| ❓ torch.distributions.Bernoulli | ⚠️ | ✅ | ✅ | ✅ |
| ❓ torch.distributions.Categorical | ⚠️ | ✅ | ✅ | ✅ |
| torch.distributions.ExpTransform | ❌ | ❌ | ✅ | ✅ |
| ❓ torch.distributions.Geometric | ⚠️ | ✅ | ✅ | ✅ |
| ❓ torch.distributions.Multinomial | ⚠️ | ✅ | ✅ | ✅ |
| torch.distributions.PowerTransform | ❌ | ❌ | ❌ | ✅ |
| 🟢 torch.from_numpy | ✅ | ❌ | ✅ | ✅ |
| torch.histogramdd | ✅ | ✅ | ✅ | ❌ |
| torch.inference_mode | ✅ | ❌ | ✅ | ⚠️ |
| 🟢 torch.linalg.matrix_rank | ✅ | ❌ | ❌ | ✅ |
| 🟢 torch.linalg.solve_triangular | ✅ | ✅ | ✅ | ❌ |
| ❓ torch.load | ✅ | ❌ | ✅ | ✅ |
| 🔁 torch.max | ✅ | ❌ | ✅ | ⚠️ |
| ❓ torch.max_pool1d | ⚠️ | ✅ | ✅ | ✅ |
| ❓ torch.max_pool2d | ⚠️ | ❌ | ❌ | ⚠️ |
| ❓ torch.max_pool3d | ⚠️ | ✅ | ❌ | ⚠️ |
| 🔁 torch.mean | ✅ | ❌ | ❌ | ✅ |
| torch.median | ❌ | ❌ | ❌ | ⚠️ |
| 🔁 torch.min | ✅ | ❌ | ✅ | ⚠️ |
| torch.nanmedian | ❌ | ❌ | ❌ | ⚠️ |
| torch.nanquantile | ✅ | ✅ | ✅ | ⚠️ |
| ❓ torch.nn.ConvTranspose1d | ⚠️ | ❌ | ✅ | ❌ |
| ❓ torch.nn.ConvTranspose2d | ⚠️ | ❌ | ✅ | ❌ |
| ❓ torch.nn.ConvTranspose3d | ⚠️ | ❌ | ✅ | ❌ |
| torch.nn.CrossEntropyLoss | ❌ | ❌ | ✅ | ✅ |
| ❓ torch.nn.Embedding | ⚠️ | ❌ | ✅ | ✅ |
| 🟢 torch.nn.GRUCell | ✅ | ✅ | ✅ | ❌ |
| 🟢 torch.nn.Identity | ✅ | ✅ | ✅ | ❌ |
| torch.nn.KLDivLoss | ❌ | ❌ | ❌ | ✅ |
| 🟢 torch.nn.LSTMCell | ✅ | ✅ | ✅ | ❌ |
| ❓ torch.nn.LogSoftmax | ✅ | ✅ | ✅ | ❌ |
| ❓ torch.nn.MaxPool1d | ⚠️ | ❌ | ❌ | ✅ |
| ❓ torch.nn.MaxPool2d | ⚠️ | ❌ | ❌ | ✅ |
| ❓ torch.nn.MaxPool3d | ⚠️ | ❌ | ❌ | ✅ |
| ❓ torch.nn.Module.register_forward_hook | ✅ | ❌ | ✅ | ✅ |
| ❓ torch.nn.Module.register_forward_pre_hook | ✅ | ❌ | ✅ | ✅ |
| ❓ torch.nn.RNNBase | ⚠️ | ❌ | ❌ | ✅ |
| ❓ torch.nn.Softmax | ✅ | ✅ | ✅ | ❌ |
| ❓ torch.nn.SyncBatchNorm.convert_sync_batchnorm | ✅ | ❌ | ✅ | ✅ |
| ❓ torch.nn.Transformer | ⚠️ | ✅ | ✅ | ✅ |
| ❓ torch.nn.TransformerDecoderLayer | ⚠️ | ❌ | ❌ | ❌ |
| ❓ torch.nn.TransformerEncoderLayer | ⚠️ | ❌ | ❌ | ❌ |
| ❓ torch.nn.Upsample | ❌ | ❌ | ❌ | ❌ |
| torch.nn.UpsamplingBilinear2d | ❌ | ❌ | ❌ | ❌ |
| torch.nn.UpsamplingNearest2d | ❌ | ❌ | ❌ | ❌ |
| ❓ torch.nn.functional.embedding | ⚠️ | ❌ | ❌ | ✅ |
| torch.nn.functional.grid_sample | ❌ | ❌ | ❌ | ✅ |
| ❓ torch.nn.functional.interpolate | ❌ | ❌ | ❌ | ❌ |
| torch.nn.functional.kl_div | ❌ | ❌ | ✅ | ✅ |
| ❓ torch.nn.functional.log_softmax | ❌ | ❌ | ❌ | ❌ |
| ❓ torch.nn.functional.max_pool1d | ⚠️ | ❌ | ❌ | ❌ |
| ❓ torch.nn.functional.max_pool2d | ⚠️ | ❌ | ❌ | ❌ |
| ❓ torch.nn.functional.max_pool3d | ⚠️ | ✅ | ✅ | ❌ |
| ❓ torch.nn.functional.max_unpool1d | ❌ | ❌ | ❌ | ❌ |
| torch.nn.functional.pad | ✅ | ✅ | ✅ | ❌ |
| ❓ torch.nn.functional.softmax | ❌ | ❌ | ❌ | ❌ |
| torch.nn.functional.softmin | ❌ | ✅ | ❌ | ❌ |
| 🟢 ❓ torch.nn.functional.upsample | ❌ | ❌ | ❌ | ❌ |
| 🟢 torch.nn.functional.upsample_bilinear | ❌ | ❌ | ❌ | ❌ |
| 🟢 torch.nn.functional.upsample_nearest | ❌ | ❌ | ❌ | ❌ |
| torch.nn.init.xavier_normal_ | ❌ | ✅ | ❌ | ✅ |
| torch.nn.init.xavier_uniform_ | ❌ | ✅ | ❌ | ✅ |
| torch.nn.utils.vector_to_parameters | ✅ | ❌ | ❌ | ✅ |
| torch.nonzero | ✅ | ❌ | ❌ | ⚠️ |
| ❓ torch.norm | ✅ | ❌ | ❌ | ❌ |
| 🔁 torch.normal | ❌ | ❌ | ❌ | ❌ |
| ❓ torch.optim.Adagrad | ⚠️ | ✅ | ✅ | ✅ |
| ❓ torch.optim.SGD | ⚠️ | ✅ | ✅ | ❌ |
| ❓ torch.optim.lr_scheduler.CyclicLR | ⚠️ | ✅ | ✅ | ✅ |
| ❓ torch.optim.lr_scheduler.OneCycleLR | ⚠️ | ✅ | ✅ | ✅ |
| 🔁 torch.prod | ❌ | ❌ | ❌ | ✅ |
| 🟢 torch.profiler.schedule | ✅ | ✅ | ✅ | ❌ |
| torch.quantile | ✅ | ✅ | ✅ | ⚠️ |
| 🔁 torch.scatter | ✅ | ❌ | ❌ | ❌ |
| 🔁 torch.sort | ✅ | ❌ | ❌ | ✅ |
| ❓ torch.sparse.sum | ❌ | ✅ | ❌ | ❌ |
| 🟢 torch.sparse_csr_tensor | ✅ | ✅ | ✅ | ❌ |
| 🔁 torch.std | ✅ | ❌ | ❌ | ❌ |
| 🔁 torch.std_mean | ✅ | ❌ | ❌ | ✅ |
| ❓ torch.svd | ❌ | ❌ | ❌ | ⚠️ |
| 🔁 torch.trapezoid | ❌ | ❌ | ✅ | ⚠️ |
| 🔁 torch.trapz | ❌ | ❌ | ✅ | ⚠️ |
| ❓ torch.unique | ⚠️ | ✅ | ✅ | ✅ |
| 🟢 torch.utils.cpp_extension.CUDAExtension | ✅ | ❌ | ❌ | ❌ |
| 🟢 torch.utils.cpp_extension.CppExtension | ✅ | ❌ | ❌ | ❌ |
| ❓ torch.utils.data.DataLoader | ⚠️ | ❌ | ✅ | ✅ |
| 🟢 torch.utils.dlpack.to_dlpack | ✅ | ❌ | ✅ | ⚠️ |
| 🟢 torch.var | ✅ | ❌ | ❌ | ✅ |
| 🔁 torch.var_mean | ✅ | ❌ | ❌ | ✅ |

### PR APIs

```bash

```

### CODE REVIEW

**Comment by zhwesky2010** [comment]
@RedContritio 修复一下CI

**Comment by zhwesky2010** [comment]
@RedContritio CI还是失败了

**Comment by RedContritio** [comment]
> @RedContritio CI还是失败了

本地测试正常，CI 过不了，我添加 log 试试

File: paconvert/api_mapping.json [hunk 1]
```diff
@@ -7161,7 +7168,8 @@
     "paddle_api": "paddle.to_tensor",
     "min_input_args": 1,
     "args_list": [
-      "ndarray"
+      "ndarray",
+      "/"
```
Comment (zhwesky2010) [inline paconvert/api_mapping.json:7172]: 这个不对吧

File: paconvert/api_matcher.py [hunk 2]
```diff
@@ -1182,7 +1182,7 @@ def generate_code(self, kwargs):
 class TensorMatcher(BaseMatcher):
     def get_paddle_api(self):
         if isinstance(
-            self.transformer.parent_node, (ast.Tuple, ast.arg, ast.Subscript)
+            self.transformer.parent_node, (ast.Tuple, ast.Index, ast.arg, ast.Subscript)
```
Comment (zhwesky2010) [inline paconvert/api_matcher.py:1185]: 这个是什么case

File: tests/test_Tensor_contiguous.py [hunk 3]
```diff
@@ -40,3 +40,16 @@ def test_case_2():
         """
     )
     obj.run(pytorch_code, ["result"])
+
+
+# Convert torch.contiguous_format to Paddle is not supported currently
```
Comment (zhwesky2010) [inline tests/test_Tensor_contiguous.py:45]: 这个参数直接移除是不是就行，GenericMatcher中针对memory_format的通用处理就是移除，在映射文档里的原则也是删除

File: tests/test_Tensor_is_contiguous.py [hunk 4]
```diff
@@ -49,3 +49,15 @@ def test_case_3():
...
     obj.run(pytorch_code, ["result"])
+
+
+# Convert torch.contiguous_format to Paddle is not supported currently
+def _test_case_4():
+    pytorch_code = textwrap.dedent(
+        """
+        import torch
+        a = torch.tensor([[4, 9], [23, 2]], dtype=torch.complex128)
+        result = a.is_contiguous(memory_format=torch.contiguous_format)
```
Comment (zhwesky2010) [inline tests/test_Tensor_is_contiguous.py:60]: 同上

File: tools/consistency/consistency_check.py [hunk 5]
```diff
@@ -51,7 +53,15 @@ def _compare_content(actual_dir, expect_dir):
                 content1,
             )
             if content1 != content2:
-
+                # print inconsistent content
```
Comment (zhwesky2010) [inline tools/consistency/consistency_check.py:56]: 这个是调试信息还是？需要保留吗

File: tools/validate_unittest/validate_unittest.py [hunk 6]
```diff
@@ -377,6 +378,9 @@ def check_call_variety(test_data, api_mapping, *, api_alias={}, verbose=True):
                 if arg == var_args_collector:
                     args_list_full_positional.append(arg)
                     var_arg_name = arg
+            elif arg == "/":
```
Comment (zhwesky2010) [inline tools/validate_unittest/validate_unittest.py:381]: 用这个 /  是分隔什么？最好不改json，避免影响通用的开发规则

File: paconvert/api_mapping.json (hunk 1 above)
Comment (RedContritio) [reply paconvert/api_mapping.json:7172]: 这里因为这个参数只能以位置参数传入，不接受关键字形式，所以这样标注了

```python
def func_with_positional_keyword_args(param1, /, keyword1=None):
    print(param1, keyword1)

func_with_positional_keyword_args(10)
```

File: paconvert/api_matcher.py (hunk 2 above)
Comment (RedContritio) [reply paconvert/api_matcher.py:1185]: 这个是 `tests/code_library/code_case/torch_code/type_hinting.py`

```python
def forward(
        hidden_states: Optional[Tuple[torch.FloatTensor]],
        rotary_pos_emb_list: Optional[List[List[torch.Tensor]]] = None,
        layer_past: Optional[Tuple[torch.Tensor]] = None,
        attention_mask: Optional[torch.FloatTensor] = None,
        head_mask: Optional[torch.FloatTensor] = None,
        encoder_hidden_states: Optional[torch.Tensor] = None,
        encoder_attention_mask: Optional[torch.FloatTensor] = None,
        output_attentions: Optional[bool] = False,
        use_cache: Optional[bool] = False,
    ):
    pass
```

其中的 `Tuple[torch.FloatTensor]` 会被解释成 `ast.Index`

**Review by zhwesky2010 — APPROVED** [review]
